Userland/Libraries/LibWeb/DOM/RadioNodeList.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/Bindings/RadioNodeListPrototype.h>
+#include <LibWeb/DOM/Element.h>
+#include <LibWeb/DOM/RadioNodeList.h>
+
+namespace Web::DOM {
+
+JS::NonnullGCPtr<RadioNodeList> RadioNodeList::create(JS::Realm& realm, Vector<JS::Handle<Element>> const& elements)
+{
+    return realm.heap().allocate<RadioNodeList>(realm, realm, elements);
+}
+
+RadioNodeList::RadioNodeList(JS::Realm& realm, Vector<JS::Handle<Element>> const& nodes)
+    : NodeList(realm)
+    , m_node_list(nodes)
+{
+}
+
+RadioNodeList::~RadioNodeList() = default;
+
+void RadioNodeList::initialize(JS::Realm& realm)
+{
+    Base::initialize(realm);
+    set_prototype(&Bindings::ensure_web_prototype<Bindings::RadioNodeListPrototype>(realm, "RadioNodeList"));
+}
+
+// https://dom.spec.whatwg.org/#dom-nodelist-length
+u32 RadioNodeList::length() const
+{
+    // The length attribute must return the number of nodes represented by the collection.
+    return m_node_list.size();
+}
+
+// https://dom.spec.whatwg.org/#dom-nodelist-item
+Node const* RadioNodeList::item(u32 index) const
+{
+    // The item(index) method must return the indexth node in the collection. If there is no indexth node in the collection, then the method must return null.
+    if (index >= m_node_list.size())
+        return nullptr;
+    return static_cast<ParentNode const*>(m_node_list[index].ptr());
```

Comment:
This `static_cast` doesn't really feel correct here, I feel like I may be using wrong approach for this?